src/api/status/public/assets/js/gitHubStats.js

```diff
@@ -0,0 +1,80 @@
+const fetchGitHubData = (repo) => {
+  let weeklyCommits = 0;
+
+  // get weekly commits for last year: https://docs.github.com/en/rest/reference/repos#get-the-weekly-commit-count
+  fetch(`https://api.github.com/repos/Seneca-CDOT/${repo}/stats/participation`, {
```

Comment:
Write a function to build your GitHub API urls for you vs. hard-coding below:

```
const ghApiUrl = (repo = 'Seneca-CDOT', path) => `https://api.github.com/repos/Seneca-CDOT/${repo}{path}`
```

Then you can use it like `fetch(ghApiUrl('/stats/participation))`

Reply:
Reading further down, you can actually reuse a bunch of code here:

```js
const fetchGitHubApi = (repo = 'Seneca-CDOT', path) =>
  fetch(`https://api.github.com/repos/${repo}{path}`, {
    headers: {  Accept: 'application/vnd.github.v3+json', }
  })
  .then(res => {
    if(!res.ok) {
      throw new Error(`unable to get ${path}`);
    }
    return res.json()
  )
  .catch(err => console.error(err));
```

Now you can do:

```js
fetchGitHubApi('/stats/participation')
  .then(data => { /* process data here */);
```

Reply:
yes that's much better!

Reply:
I couldn't reuse some parts because I don't always return `res.json()`.